recipe-server/normandy/recipes/tests/test_api.py

```diff
@@ -415,6 +416,71 @@ def test_it_serves_revisions(self, api_client):
         assert res.status_code == 200
         assert res.data['id'] == recipe.latest_revision.id
 
+    def test_request_approval(self, api_client):
+        recipe = RecipeFactory()
+        res = api_client.post(
+            '/api/v1/recipe_revision/{}/request_approval/'.format(recipe.latest_revision.id))
+        assert res.status_code == 200
+        assert res.data['id'] == recipe.latest_revision.approval_request.id
+
+    def test_cannot_open_second_approval_request(self, api_client):
+        recipe = RecipeFactory()
+        ApprovalRequestFactory(revision=recipe.latest_revision)
+        res = api_client.post(
+            '/api/v1/recipe_revision/{}/request_approval/'.format(recipe.latest_revision.id))
+        assert res.status_code == 400
+
+
+@pytest.mark.django_db
+class TestApprovalRequestAPI(object):
+    def test_it_works(self, api_client):
+        res = api_client.get('/api/v1/approval_request/')
+        assert res.status_code == 200
+        assert res.data == []
+
+    def test_approve(self, api_client):
+        r = RecipeFactory()
+        a = ApprovalRequestFactory(revision=r.latest_revision)
+        res = api_client.post('/api/v1/approval_request/{}/approve/'.format(a.id))
+        assert res.status_code == 200
+
+        r = Recipe.objects.get(pk=r.pk)
+        assert r.is_approved
+
+    def test_approve_already_approved(self, api_client):
+        r = RecipeFactory()
+        a = ApprovalRequestFactory(revision=r.latest_revision)
+        a.approve(UserFactory())
+
+        res = api_client.post('/api/v1/approval_request/{}/approve/'.format(a.id))
+        assert res.status_code == 400
```

Comment:
It is probably a good idea to assert the text of the error that was returned here.